File: DecompiledGameScripts/core/gameplay/prereqs/districtPrereq.swift

public class DistrictPrereq extends IScriptablePrereq {

  private let m_district: wref<District_Record>;

  protected func Initialize(recordID: TweakDBID) -> Void {
    let record: ref<DistrictPrereq_Record> = TweakDBInterface.GetDistrictPrereqRecord(recordID);
    this.m_district = record.District();
  }

  public const func IsFulfilled(game: GameInstance, context: ref<IScriptable>) -> Bool {
    let ps: ref<PreventionSystem> = GameInstance.GetScriptableSystemsContainer(game).Get(n"PreventionSystem") as PreventionSystem;
    return ps.GetCurrentDistrict().GetDistrictID() == this.m_district.GetRecordID();
  }
}

File: Definitions/Game_Definitions/classes/DistrictPrereq.lua
---@meta
---@diagnostic disable

---@class DistrictPrereq : gameIScriptablePrereq
---@field district gamedataDistrict_Record
DistrictPrereq = {}

---@return DistrictPrereq
function DistrictPrereq.new() return end

---@param props table
---@return DistrictPrereq
function DistrictPrereq.new(props) return end

---@param recordID TweakDBID|string
function DistrictPrereq:Initialize(recordID) return end

---@param context IScriptable
---@return Bool
function DistrictPrereq:IsFulfilled(context) return end

